### PR TITLE
https://github.com/CODAIT/stocator/issues/209 fix + checkstyle fix

### PR DESCRIPTION
**Pull request by michaelaltar**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by gilv** [comment]
@michaelaltar thanks...how  this resolve the issue in #209?
Can you please resolve conflict in the pull request?

**Comment by michaelaltar** [comment]
@gilv This branch has no conflicts. Ci build is failed. I'm compiling with maven and everything is OK.
This error happens when map have collision, but it's not thread-safety, so using concurrent hash map is solves this issue for us.

**Comment by gilv** [comment]
@michaelaltar thanks

**Comment by gilv** [comment]
@michaelaltar can you please submit this pull request against master branch? I will then merge your commit to cos-sdk branch as well.

File: src/main/java/com/ibm/stocator/fs/cos/COSAPIClient.java [hunk 1]
```diff
@@ -227,7 +228,7 @@
    * Contains map of object names that were created. Used in container
    * listing
    */
-  private Map<String, Boolean> mCachedSparkOriginated;
+  private ConcurrentHashMap<String, Boolean> mCachedSparkOriginated;
```
Comment (cmuchinsky) [inline src/main/java/com/ibm/stocator/fs/cos/COSAPIClient.java]: Style nit, declaration should probably remain an interface, either leave as `Map`, or switch to `ConcurrentMap` if any concurrency specific methods are called

**Comment by gilv** [comment]
@michaelaltar can you please address the comments so we can merge it?

File: src/main/java/com/ibm/stocator/fs/cos/COSAPIClient.java (hunk 1 above)
Comment (michaelaltar) [reply src/main/java/com/ibm/stocator/fs/cos/COSAPIClient.java]: open new merge request to master as @gilv  ask. https://github.com/CODAIT/stocator/pull/213

**Comment by michaelaltar** [comment]
open new request https://github.com/CODAIT/stocator/pull/213 , so u can close one

**Comment by michaelaltar** [comment]
open new request https://github.com/CODAIT/stocator/pull/213 , so u can close one

**Pull request closed without merging**